Dockerfile

```diff
@@ -1,31 +1,55 @@
-FROM rustlang/rust:nightly AS builder
-RUN apt update && apt install -y npm && rm -rf /var/apt/lists
-ARG GITHUB_SHA
-ENV GITHUB_SHA=$GITHUB_SHA
-ARG GITHUB_REPOSITORY
-ENV GITHUB_REPOSITORY=$GITHUB_REPOSITORY
+# Setup chef
+FROM rustlang/rust:nightly AS chef
+RUN cargo install cargo-chef
+
+# Build
+FROM chef AS planner
 WORKDIR /app
-COPY Cargo.toml Cargo.lock package.json ./
-RUN npm i
-RUN mkdir src && echo "fn main(){println!(\"dummy\");}" > ./src/main.rs && touch src/lib.rs
-RUN mkdir src/bin && echo "fn main(){println!(\"dummy\");}" > ./src/bin/bell_mkschema.rs
-RUN cargo build --release --bin ethsbell-rewrite --features=ws
-RUN rm ./src/main.rs && rm ./src/lib.rs
-COPY . .
-RUN touch src/main.rs && touch src/lib.rs
-# This is here so the build fails if the tests do.
+COPY Cargo.toml Cargo.lock ./
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+WORKDIR /app
+COPY --from=planner /app/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+
+# Build image and run tests
+COPY LICENSE.html def.json .
```

Comment:
This step fails because you removed the default def.json. It is also a lint warning: when copying multiple files, the target directory should always end with a slash.

Reply:
Fixed